client/my-sites/themes/theme-showcase.jsx

```diff
@@ -295,7 +309,7 @@ class ThemeShowcase extends React.Component {
 							className="themes__upload-button"
 							compact
 							onClick={ this.onUploadClick }
-							href={ siteSlug ? `/themes/upload/${ siteSlug }` : '/themes/upload' }
+							href={ getInstallThemeSlug( siteSlug, canUploadThemesOrPlugins ) }
```

Comment:
Isn't the redirection enough we do on the `Upload` component enough?
https://github.com/Automattic/wp-calypso/pull/52776/files#diff-579ae2c0c2c5ce152deb47027c130c06f93fabdf53b59dac04d8b15a69d60745R96-R98

Reply:
It would work also, but I prefer to leave this because:

* We can avoid the user loading the additional view, when we are just going to redirect them. If we know their destination ahead of time the better.
* The intent feels more deliberate here too so anyone reading it will know what to expect.